src/b-splines/indexing.jl

```diff
@@ -221,3 +219,17 @@ function symmatrix(h::NTuple{L,Any}) where L
         SMatrix{N,N}(h[i] for i in l)
     end
 end
+
+# Hardcode the output size.
+for N in 4:7
+    L = N*(N+1)÷2
+    @eval symsize(::Val{$L}) = $N
```

Comment:
Why not use [`@generated`](https://docs.julialang.org/en/v1/manual/metaprogramming/#Generated-functions) here rather than `@eval`?

Reply:
Do you mean:
```julia
@generated function f(::Val{L}) where {L}
           N = floor(Int, sqrt(2L))
           N*(N+1) == 2L || error()
           return :($N)
       end
```
?

Reply:
That looks good if it works.